File: examples/cpp/Example00HelloWorld.cpp

#include "maf/maf.hpp"

#include <mpi.h>

class HelloWorldAction : public maf::Action, public maf::Transmittable<HelloWorldAction> {

public:

    HelloWorldAction() : maf::Action() {

    };

    void run() {
        maf::log("Hello World! (from c++)");
    };

};

int main(int argc, char* argv[]) {

    try {
        maf::ActionFactory::Register<HelloWorldAction>();

        std::vector<std::shared_ptr<maf::Action>> actions;
        actions.push_back(std::shared_ptr<maf::Action>(new HelloWorldAction));
        maf::BcastController controller(actions);
        controller.start();
    }
    catch (std::exception& ex) {
        maf::log("FAILED: ", ex.what());
        MPI_Abort(MPI_COMM_WORLD, -1);
    }
    catch (...) {
        maf::log("FAILED: no idea what happened");
        MPI_Abort(MPI_COMM_WORLD, -1);
    }
}

File: examples/py/example00_hello_world.py
'''Simple queue bcast controller'''
import maf


class AnnoyContext(maf.Context):

    def __init__(self):
        maf.Context.__init__(self)
        self.annoyance_count = 0

    def annoy(self):
        self.annoyance_count += 1
        if self.annoyance_count > 2:
            maf.log("Ugh, so annoyed!!! {}".format(self.annoyance_count))
        else:
            maf.log("Nope, not annoyed... {}".format(self.annoyance_count))

@maf.action
class HelloWorldAction(maf.Action):

    def run(self):
        maf.log('Hello World! (from Python) there are {} actions in the stack'.format(len(maf._ACTION_STACK)))

actions = [HelloWorldAction()]
controller = maf.BcastController(actions)
controller.start(AnnoyContext())
